ballerina/openapi_utils.bal

```
@@ -224,10 +224,16 @@ class OpenApiSpecVisitor {
                 schema: check self.visitSchema(schema)
             };
         }
-        if schema is Reference {
-            string refName = regexp:split(re `/`, schema.\$ref).pop();
-            schema = {'type: OBJECT, properties: {[refName] : schema}};
+        string? xmlName = schema.'xml?.name;
+        string refName;
+        if xmlName is string {
+            refName = xmlName;
+        } else if schema is Reference {
+            refName = regexp:split(re `/`, schema.\$ref).pop();
+        } else {
+            return error("Schema should have a name for xml content type.", 'schema = schema);
```

what will happen when schema does not have the xml name? is it an invalid schema according to OpenAPI? only if so we can throw an error here.

---

Then use the correct error time (e.g. OpenAPIParseError)